pkg/runs/runsDownload.go

```diff
@@ -29,24 +30,75 @@ func DownloadArtifacts(
 
 	var err error = nil
 	var runs []galasaapi.Run
-	var artifactPaths []string
 
 	runs, err = GetRunsFromRestApi(runName, 0, 0, timeService, apiServerUrl)
 	if err == nil {
-		for _, run := range runs {
-			runId := run.GetRunId()
-			artifactPaths, err = GetArtifactPathsFromRestApi(runId, apiServerUrl)
-			if err == nil {
-				for _, artifactPath := range artifactPaths {
-					var artifactData io.Reader
-					artifactData, err = GetFileFromRestApi(runId, strings.TrimPrefix(artifactPath, "/"), apiServerUrl)
-					if err == nil {
-						err = WriteArtifactToFileSystem(fileSystem, runName, artifactPath, artifactData, forceDownload)
-					}
+		if len(runs) > 1 {
+			// get list of runs that are reRuns - get list of runs that are reRuns of each other
+			// create a map of lists of reRuns - key is queued time, value is the run
+
+			reRunsByQueuedTime := make(map[string][]galasaapi.Run, 0)
+			queuedTime := ""
+			for _, run := range runs {
+				queuedTime = run.TestStructure.GetQueued()
+				runsWithSameQueuedTime := reRunsByQueuedTime[queuedTime]
+				runsWithSameQueuedTime = append(runsWithSameQueuedTime, run)
+				reRunsByQueuedTime[queuedTime] = runsWithSameQueuedTime
+			}
+
+			// SORTING BY START TIME TO DO - on api side not in cli
+
+			for _, reRunsList := range reRunsByQueuedTime {
+				for count, reRun := range reRunsList {
+					directoryName := nameArtifactDownloadDirectory(reRun, count)
+					err = downloadArtifactsToDirectory(apiServerUrl, directoryName, reRun, fileSystem, forceDownload)
 				}
 			}
+
+		} else {
+			for _, run := range runs {
```

Comment:
Loop should fail on first error it gets. Else the error is over-written and lost. Loop needs to break out.